src/MizarWidgets/SamplingWithFrameTrackReportConfigValidatorTest.cpp

```diff
@@ -0,0 +1,115 @@
+
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "MizarBase/BaselineOrComparison.h"
+#include "MizarWidgets/SamplingWithFrameTrackReportConfigValidator.h"
+#include "QString"
+
+using ::orbit_mizar_base::Baseline;
+using ::orbit_mizar_base::Comparison;
+using ::orbit_mizar_base::TID;
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+
+namespace {
+
+class MockPairedData {
+ public:
+  MOCK_METHOD(uint64_t, CaptureDuration, (), (const));
+};
+
+class MockBaselineAndComparison {
+ public:
+  MOCK_METHOD(const Baseline<MockPairedData>&, GetBaselineData, (), (const));
+  MOCK_METHOD(const Comparison<MockPairedData>&, GetComparisonData, (), (const));
+};
+
+QString last_reported_error_message;
+void MockErrorReporter(const QString& message) { last_reported_error_message = message; }
+
+}  // namespace
+
+namespace orbit_mizar_widgets {
+
+constexpr uint64_t kBaselineCaptureDuration = 123456;
+constexpr uint64_t kComparisonCaptureDuration = 234567;
+const QString kBaselineTitleStr = "Baseline";
+const QString kComparisonTitleStr = "Comparison";
+const Baseline<QString> kBaselineTitle{kBaselineTitleStr};
+const Comparison<QString> kComparisonTitle{kComparisonTitleStr};
+
+using HalfConfig = orbit_mizar_data::HalfOfSamplingWithFrameTrackReportConfig;
+
+TEST(SamplingWithFrameTrackReportConfigValidator, IsCorrect) {
+  Baseline<MockPairedData> baseline_data;
+  Comparison<MockPairedData> comparison_data;
+  MockBaselineAndComparison bac;
+
+  EXPECT_CALL(*baseline_data, CaptureDuration).WillRepeatedly(Return(kBaselineCaptureDuration));
+  EXPECT_CALL(*comparison_data, CaptureDuration).WillRepeatedly(Return(kComparisonCaptureDuration));
+
+  EXPECT_CALL(bac, GetBaselineData).WillRepeatedly(ReturnRef(baseline_data));
+  EXPECT_CALL(bac, GetComparisonData).WillRepeatedly(ReturnRef(comparison_data));
+
+  SamplingWithFrameTrackReportConfigValidatorTmpl<MockBaselineAndComparison, MockPairedData,
```

Comment:
What about marking this object `const`. (This would help to understand that the `Validate` calls below aren't actually mutating the object and the order is not relevant.)